types/xdg_shell/wlr_xdg_surface.c

```diff
@@ -337,9 +337,7 @@ static void xdg_surface_handle_surface_commit(struct wl_listener *listener,
 	}
 
 	if (surface->role == WLR_XDG_SURFACE_ROLE_NONE) {
```

Comment:
Maybe we can keep the protocol error, and replace the check `surface->surface->role == NULL` instead? This doesn't get reset by `reset_xdg_surface`.

Reply:
Fixed.